File: api/models/professor.ts
import { Schema, connection } from 'mongoose';
import { Location, Meeting } from './section'; // dependency will be resolved with section pull request

export interface Professor {
  first_name: string;
  last_name: string;
  titles: Array<string>;
  email: string;
  phone_number: string;
  office: Location;
  profile_uri: string;
  image_uri: string;
  office_hours: Array<Meeting>;
  sections: Array<Schema.Types.ObjectId>;
}

export const ProfessorSchema = new Schema<Professor>({
  first_name: { type: String, required: true },
  last_name: { type: String, required: true },
  titles: { type: [String], required: true },
  email: { type: String, required: true },
  phone_number: { type: String, required: true },
  office: { type: Object, required: true },
  profile_uri: { type: String, required: true },
  image_uri: { type: String, required: true },
  office_hours: { type: [Object], required: true },
  sections: { type: [Schema.Types.ObjectId], required: true }
});

const professorDB = connection.useDb('professorDB');
export const ProfessorModel = professorDB.model<Professor>('professor', ProfessorSchema);

File: api-graphql/graph/model/professor.go
package model

type Professor struct {
	ID          string     `json:"_id"`
	FirstName   string     `json:"first_name"`
	LastName    string     `json:"last_name"`
	Titles      []string   `json:"titles"`
	Email       string     `json:"email"`
	PhoneNumber *string    `json:"phone_number"`
	Office      *Location  `json:"office"`
	ProfileURI  *string    `json:"profile_uri"`
	ImageURI    *string    `json:"image_uri"`
	OfficeHours []*Meeting `json:"office_hours"`
	Sections    []*Section `json:"sections"`
}
